doc/api/url.md

```diff
@@ -250,7 +250,478 @@ properties of URL objects:
+
+Invalid URL characters included in the value assigned to the `password` property
+are [percent-encoded](#whatwg-percent-encoding). Note that the selection of
+which characters to percent-encode may vary somewhat from what the
+[`url.parse()`][] and [`url.format()`][] methods would produce.
+
+#### url.pathname
+
+Gets and sets the path portion of the URL.
+
+```js
+const myURL = new URL('http://example.org/abc/xyz?123');
+console.log(myURL.pathname);
+  // Prints /abc/xyz
+
+myURL.pathname = '/abcdef';
+console.log(myURL.href);
+  // Prints http://example.org/abcdef?123
+```
+
+Using the `delete` keyword (e.g. `delete myURL.pathname`) has no effect but will
+still return `true`.
+
+Invalid URL characters included in the value assigned to the `pathname`
+property are [percent-encoded](#whatwg-percent-encoding). Note that the
+selection of which characters to percent-encode may vary somewhat from what the
+[`url.parse()`][] and [`url.format()`][] methods would produce.
+
+#### url.port
+
+Gets and sets the port portion of the URL. When getting the port, the value
+is returned as a String.
+
+```js
+const myURL = new URL('http://example.org:8888');
+console.log(myURL.port);
+  // Prints 8888
+
+myURL.port = 1234;
+console.log(myURL.href);
+  // Prints http://example.org:1234
+```
+
+The port value may be set as either a number or as a String containing a number
+in the range `0` to `65535` (inclusive). Setting the value to the default port
+of the `URL` objects given `protocol` will result in the `port` value becoming
+the empty string (`''`).
+
+Using the `delete` keyword (e.g. `delete myURL.port`) has no effect but will
+still return `true`.
+
+Invalid URL port values assigned to the `port` property are ignored.
+
+#### url.protocol
+
+Gets and sets the protocol portion of the URL.
+
+```js
+const myURL = new URL('http://example.org');
+console.log(myURL.protocol);
+  // Prints http:
+
+myURL.protocol = 'ftp';
+console.log(myURL.href);
+  // Prints ftp://example.org
+```
+
+Using the `delete` keyword (e.g. `delete myURL.protocol`) has no effect but will
+still return `true`.
+
+Invalid URL protocol values assigned to the `protocol` property are ignored.
+
+#### url.search
+
+Gets and sets the serialized querystring portion of the URL.
+
+```js
+const myURL = new URL('http://example.org/abc?123');
+console.log(myURL.search);
+  // Prints ?123
+
+myURL.search = 'abc=xyz';
+console.log(myURL.href);
+  // Prints http://example.org/abc?abc=xyz
+```
+
+Using the `delete` keyword (e.g. `delete myURL.search`) has no effect but will
+still return `true`.
+
+Any invalid URL characters appearing in the value assigned the `search`
+property will be [percent-encoded](#whatwg-percent-encoding). Note that the
+selection of which characters to percent-encode may vary somewhat from what the
+[`url.parse()`][] and [`url.format()`][] methods would produce.
+
+#### url.searchParams
+
+Gets a [`URLSearchParams`][#url_class_urlsearchparams] object representing the
```

Comment:
How about "Gets **the** URLSearchParams object…"

This IMO makes it clearer that the same object will be returned (c.f. the IDL `[SameObject]` attr).